dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeAPIImpl.java

```diff
@@ -395,132 +401,117 @@ private boolean fireUpdateIdentifiers(String oldVal, String newVal) {
   }
   
   @Override
-  public ContentType save(ContentType contentType, List<Field> fields) throws DotDataException, DotSecurityException {
-	  return save(contentType, fields, null);
+  public ContentType save(ContentType contentType, List<Field> newFields) throws DotDataException, DotSecurityException {
+	  return save(contentType, newFields, null);
   }
 
-  @Override
-  public ContentType save(ContentType contentType, List<Field> fields, List<FieldVariable> fieldVariables) throws DotDataException, DotSecurityException {
-	  
+    @Override
+    public ContentType save(ContentType contentType, List<Field> newFields, List<FieldVariable> newFieldVariables) throws DotDataException, DotSecurityException {
+        // Sets the host:
+        try {
+            if(contentType.host()==null){
+                contentType = ContentTypeBuilder.builder(contentType).host(Host.SYSTEM_HOST).build();
```

Comment:
This could be a constant instance final static ContentType SYSTEM_HOST_CONTENT_TYPE = ...